src/java/com/wolfssl/provider/jsse/WolfSSLEngineHelper.java

```diff
@@ -1531,6 +1538,46 @@ private boolean isLegacyDHEnabled() {
         return "legacy".equals(dhKeySize);
     }
 
+    protected synchronized boolean matchSNI(){
+        List <SNIMatcher> matchers = this.params.getSNIMatchers();
+        boolean match = false;
+        if (matchers != null && !matchers.isEmpty()) {
+            /* Match a server name to SNI requested by Client */
+            List <SNIServerName> serverNames = this.session
+                                                    .getRequestedServerNames();
+            if (serverNames != null && !serverNames.isEmpty()) {
+                for (SNIServerName serverName : serverNames) {
+                    if (serverName.getType() == WolfSSL.WOLFSSL_SNI_HOST_NAME) {
+                        /* If the SNI is of type WOLFSSL_SNI_HOST_NAME, compare
+                         * the name to matchers list */
+                        for (SNIMatcher matcher : matchers) {
+                            if (matcher.matches(serverName)) {
+                                /* If a match is found, accept the server name
+                                 * and return true value, break from loop */
+                                WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
+                                    "Accepted SNI: " + serverName);
+                                    match = true;
+                                    break;
+                            }
+                        }
+                    }
+                }
+            } else {
+                /* If server names are null or empty, ignore server name
+                 * indication */
+                WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
+                                    "No server names found, ignoring SNI");
```

Comment:
Minor spacing request, can you bump this line back to 4 spaces beyond the WolfSSLDebug line:

```
                WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
                    "No server names found, ignoring SNI");
```